pkg/virt-controller/watch/migration.go

```diff
@@ -462,6 +462,20 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 			LastProbeTime: v1.Now(),
 		}
 		migrationCopy.Status.Conditions = append(migrationCopy.Status.Conditions, condition)
+	} else if migration.DeletionTimestamp != nil && !migration.IsFinal() &&
+		conditionManager.HasCondition(migration, virtv1.VirtualMachineInstanceMigrationAbortRequested) {
+
+		targetPods, err := c.listMatchingTargetPods(migration, vmi)
```

Comment:
AFAIK, if the migration is deleted and the VMI's status says that the VMI is not in a migration process and runs on node A - if a virt-launcher pod starts running on node B it would be cleaned up by the relevant virt-handler that would recognize it as an orphaned pod that does not relate to any VMIs.

So I think that in this corner case we would waste some resources on scheduling and starting the virt-launcher pod, but it would still be safe.

Do you have any thoughts on this? @vladikr

Reply:
The migration controller is responsible for removing the target pod. It needs sync but here the migration will be removed so I think this will not happen. I am not sure what routine in virt-handler you are referring to. Running Pod would clean up itslef by timing out but not stuck one.

Even if the above would be false I found it confusing and I don't understand why the whole `if branch` is needed.  I am against adding more cases here. Why can't we simply finalize the migration and follow the normal workflow?

One last thing is that what you described seems to be contradicting one of the goals of PR. It is not only a matter of correctness but also matters of how fast I can create new migrations or why did it actually not clean up properly?

Reply:
@vladikr @xpivarc 
So let me see if I got it right - you think that the virt handler (probably in `handleMigrationAbort()` function) will remove the VMIM's finalizer and mark it as aborted to the controller would continue with its regular flow and clean up the VMIM?

The reason I did not follow that approach is that virt-handler's `VirtualMachineController` does not currently deal with VMIMs at all. It doesn't have a VMIM informer for example. This seemed to me like crossing the boundaries of virt-handler's responsibilities. But your suggested approach makes sense.

WDYT the best approach here is? Introducing a new informer to virt-handler's `VirtualMachineController`? or simply making an API call?